dtls_debug.c

```diff
@@ -127,31 +138,43 @@ dsrv_print_addr(const session_t *addr, char *buf, size_t len) {
       return 0;
 
     *p++ = '[';
+    --len;
 
     addrptr = &addr->addr.sin6.sin6_addr;
     port = ntohs(addr->addr.sin6.sin6_port);
 
     break;
   default:
-    memcpy(buf, "(unknown address type)", min(22, len));
-    return min(22, len);
+    // include terminating \000
+    append = min(23, len);
+    memcpy(p, "(unknown address type)", append);
+    if (len < 23) {
+      p[len - 1] = 0;
+    }
+    return append;
   }
 
   if (inet_ntop(addr->addr.sa.sa_family, addrptr, p, len) == 0) {
     perror("dsrv_print_addr");
     return 0;
   }
 
-  p += dtls_strnlen(p, len);
+  // append inet_ntop to p
```

Comment:
Please use the `/* ... */` comment syntax.

Reply:
Done